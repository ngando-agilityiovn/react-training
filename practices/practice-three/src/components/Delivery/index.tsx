import Bag from '@/components/Icons/bag';
import Car from '@/components/Icons/car';
import { Box, Flex, Text } from '@chakra-ui/react';

// Icon components

interface IDeliveryItem {
  title: string;
  description: string;
}

interface IDelivery {
  data: IDeliveryItem[];
}

const Delivery = ({ data }: IDelivery) => {
  return (
    <Box
      p="17px"
      borderRadius="14px"
      border="1px solid"
      borderColor="gainsboro"
      w="582px"
    >
      {data.map(({ title, description }) => (
        <>
          <Flex gap="14px">
            <Box>{title === 'Free Delivery' ? <Car /> : <Bag />}</Box>
            <Box key={title}>
              <Text
                mb="7px"
                color="secondary"
                fontWeight="bold"
                fontSize="17px"
              >
                {title}
              </Text>
              <Text variant="secondary">{description}</Text>
            </Box>
          </Flex>
          <Box my="21px" w="548px" h="1px" background="gainsboro"></Box>
        </>
      ))}
    </Box>
  );
};

export default Delivery;
