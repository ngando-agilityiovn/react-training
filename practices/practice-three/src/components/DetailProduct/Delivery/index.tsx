import Bag from '@/components/Icons/bag'
import Car from '@/components/Icons/car'
import { Box, Flex, Text } from '@chakra-ui/react'

// Icon components

interface IDeliveryItem {
  title: string
  description: string
}

interface IDelivery {
  deliveryDate: IDeliveryItem[]
}

const Delivery = ({ deliveryDate }: IDelivery) => {
  return (
    <Box
      p="17px"
      borderRadius="14px"
      border="1px solid"
      borderColor="darkGrey40"
      w="582px"
    >
      {deliveryDate.map(({ title, description }) => (
        <>
          <Flex gap="14px">
            <Box>{title === 'Free Delivery' ? <Car /> : <Bag />}</Box>
            <Box>
              <Text
                mb="7px"
                color="secondary"
                fontWeight="bold"
                fontSize="17px"
              >
                {title}
              </Text>
              <Text variant="secondary">{description}</Text>
            </Box>
          </Flex>
          <Box my="21px" w="548px" h="1px" background="darkGrey40"></Box>
        </>
      ))}
    </Box>
  )
}

export default Delivery
