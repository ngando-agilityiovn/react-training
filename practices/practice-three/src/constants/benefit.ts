export const BENEFIT_DATA = [
  {
    title:
      'Durable leather is easily cleanable so you can keep your look fresh.',
  },
  {
    title: 'Toe piece with star pattern adds durability.',
  },
]
