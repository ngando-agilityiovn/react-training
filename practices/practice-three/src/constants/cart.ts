export const CARD_DATA = [
  {
    id: '0',
    name: 'T-shirt',
    ratings: 5,
    price: 200,
    currency: '$',
    quantity: 2,
    reviews: [
      {
        id: 1,
        reviewer: 'John Doe',
        comments: 'Great product, very satisfied with my purchase!'
      },
      {
        id: 2,
        reviewer: 'Jane Smith',
        comments: 'Excellent service and fast delivery.'
      },
      {
        id: 3,
        reviewer: 'Bob Johnson',
        comments: 'Product was okay, but could be improved in certain areas.'
      }
    ],
    images: [
      'https://source.unsplash.com/random/587x691/?fashion',
      'https://source.unsplash.com/random/587x691/?clothes',
      'https://source.unsplash.com/random/587x691/?t-shirt',
      'https://source.unsplash.com/random/587x691/?adidas'
    ]
  },
  {
    id: '1',
    name: 'T-shirt',
    ratings: 28,
    price: 340,
    currency: '$',
    quantity: 20,
    reviews: [
      {
        id: 1,
        reviewer: 'John Doe',
        comments: 'Great product, very satisfied with my purchase!'
      },
      {
        id: 2,
        reviewer: 'Jane Smith',
        comments: 'Excellent service and fast delivery.'
      },
      {
        id: 3,
        reviewer: 'Bob Johnson',
        comments: 'Product was okay, but could be improved in certain areas.'
      }
    ],
    images: [
      'https://source.unsplash.com/random/587x691/?fashion',
      'https://source.unsplash.com/random/587x691/?clothes',
      'https://source.unsplash.com/random/587x691/?t-shirt',
      'https://source.unsplash.com/random/587x691/?adidas'
    ]
  }
]
