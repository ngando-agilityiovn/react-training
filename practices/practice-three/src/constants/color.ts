export const DATA_COLOR = [
  '#ffd3d3',
  '#ffd3f8',
  '#bbd278',
  '#ecdecc',
  '#bbc1f8'
]
