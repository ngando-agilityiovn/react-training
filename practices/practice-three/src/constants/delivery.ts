export const DELIVERY_DATA = [
  {
    title: 'Free Delivery',
    description: 'Enter your Postal code for Delivery Availability',
  },
  {
    title: 'Return Delivery',
    description: 'Free 30 days Delivery Return. Details',
  },
]
