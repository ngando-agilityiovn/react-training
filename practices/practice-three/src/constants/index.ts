export * from './navbar'
export * from './filter'
