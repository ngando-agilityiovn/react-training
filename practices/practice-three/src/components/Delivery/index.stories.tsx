import { Meta, StoryObj } from '@storybook/react'
import { BrowserRouter } from 'react-router-dom'

// Constants
import { DELIVERYDATA } from '@/constants'

// Components
import Delivery from '.'

export default {
  title: 'Components/Delivery',
  component: Delivery,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
} as Meta

type Story = StoryObj<typeof Delivery>

export const Catogory: Story = {
  args: {
    deliveryDate: DELIVERYDATA,
  },
}
