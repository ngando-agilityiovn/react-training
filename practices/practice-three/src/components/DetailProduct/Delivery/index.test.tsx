import { render } from '@testing-library/react'

// Constants
import { DELIVERY_DATA } from '@/constants'

// Components
import Delivery from '.'

describe('Delivery component', () => {
  it('Render correcty', () => {
    const container = render(<Delivery deliveryDate={DELIVERY_DATA} />)
    expect(container).toMatchSnapshot()
  })
})
