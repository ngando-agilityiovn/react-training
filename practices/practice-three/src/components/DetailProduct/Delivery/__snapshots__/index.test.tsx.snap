// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delivery component Render correcty 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-11439nx"
      >
        <div
          class="css-j5eaa4"
        >
          <div
            class="css-0"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="Frame"
              >
                <path
                  d="M15.75 10.5V6C15.75 5.00544 15.3549 4.05161 14.6516 3.34835C13.9484 2.64509 12.9945 2.25 12 2.25C11.0054 2.25 10.0516 2.64509 9.34833 3.34835C8.64507 4.05161 8.24998 5.00544 8.24998 6V10.5M19.606 8.507L20.869 20.507C20.939 21.172 20.419 21.75 19.75 21.75H4.24998C4.09219 21.7502 3.93613 21.7171 3.79193 21.6531C3.64774 21.589 3.51863 21.4953 3.41301 21.3781C3.30738 21.2608 3.2276 21.1227 3.17884 20.9726C3.13008 20.8226 3.11343 20.6639 3.12999 20.507L4.39399 8.507C4.42314 8.23056 4.55361 7.9747 4.76024 7.78876C4.96686 7.60281 5.23501 7.49995 5.51298 7.5H18.487C19.063 7.5 19.546 7.935 19.606 8.507ZM8.62498 10.5C8.62498 10.5995 8.58548 10.6948 8.51515 10.7652C8.44482 10.8355 8.34944 10.875 8.24998 10.875C8.15053 10.875 8.05515 10.8355 7.98482 10.7652C7.91449 10.6948 7.87498 10.5995 7.87498 10.5C7.87498 10.4005 7.91449 10.3052 7.98482 10.2348C8.05515 10.1645 8.15053 10.125 8.24998 10.125C8.34944 10.125 8.44482 10.1645 8.51515 10.2348C8.58548 10.3052 8.62498 10.4005 8.62498 10.5ZM16.125 10.5C16.125 10.5995 16.0855 10.6948 16.0152 10.7652C15.9448 10.8355 15.8494 10.875 15.75 10.875C15.6505 10.875 15.5551 10.8355 15.4848 10.7652C15.4145 10.6948 15.375 10.5995 15.375 10.5C15.375 10.4005 15.4145 10.3052 15.4848 10.2348C15.5551 10.1645 15.6505 10.125 15.75 10.125C15.8494 10.125 15.9448 10.1645 16.0152 10.2348C16.0855 10.3052 16.125 10.4005 16.125 10.5Z"
                  id="Vector"
                  stroke="#d75951"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </g>
            </svg>
          </div>
          <div
            class="css-0"
          >
            <p
              class="chakra-text css-omcqsq"
            >
              Free Delivery
            </p>
            <p
              class="chakra-text css-0"
            >
              Enter your Postal code for Delivery Availability
            </p>
          </div>
        </div>
        <div
          class="css-7lfzx5"
        />
        <div
          class="css-j5eaa4"
        >
          <div
            class="css-0"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="Frame"
              >
                <path
                  d="M8.25002 18.75C8.25002 19.1478 8.09198 19.5293 7.81068 19.8107C7.52937 20.092 7.14784 20.25 6.75002 20.25C6.35219 20.25 5.97066 20.092 5.68936 19.8107C5.40805 19.5293 5.25002 19.1478 5.25002 18.75M8.25002 18.75C8.25002 18.3522 8.09198 17.9706 7.81068 17.6893C7.52937 17.408 7.14784 17.25 6.75002 17.25C6.35219 17.25 5.97066 17.408 5.68936 17.6893C5.40805 17.9706 5.25002 18.3522 5.25002 18.75M8.25002 18.75H14.25M5.25002 18.75H3.37502C3.07665 18.75 2.7905 18.6315 2.57952 18.4205C2.36855 18.2095 2.25002 17.9234 2.25002 17.625V14.25M14.25 18.75H16.5M14.25 18.75V14.25M2.25002 14.25V6.61499C2.24843 6.34116 2.34848 6.07648 2.53081 5.87217C2.71313 5.66786 2.96477 5.53845 3.23702 5.50899C6.57006 5.16302 9.92997 5.16302 13.263 5.50899C13.828 5.56699 14.25 6.04699 14.25 6.61499V7.57299M2.25002 14.25H14.25M19.5 18.75C19.5 19.1478 19.342 19.5293 19.0607 19.8107C18.7794 20.092 18.3978 20.25 18 20.25C17.6022 20.25 17.2207 20.092 16.9394 19.8107C16.6581 19.5293 16.5 19.1478 16.5 18.75M19.5 18.75C19.5 18.3522 19.342 17.9706 19.0607 17.6893C18.7794 17.408 18.3978 17.25 18 17.25C17.6022 17.25 17.2207 17.408 16.9394 17.6893C16.6581 17.9706 16.5 18.3522 16.5 18.75M19.5 18.75H20.625C21.246 18.75 21.754 18.246 21.715 17.626C21.5141 14.3236 20.4019 11.1416 18.502 8.43299C18.3211 8.17927 18.0848 7.97001 17.8111 7.82102C17.5373 7.67203 17.2333 7.58722 16.922 7.57299H14.25M14.25 7.57299V14.25"
                  id="Vector"
                  stroke="#d75951"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </g>
            </svg>
          </div>
          <div
            class="css-0"
          >
            <p
              class="chakra-text css-omcqsq"
            >
              Return Delivery
            </p>
            <p
              class="chakra-text css-0"
            >
              Free 30 days Delivery Return. Details
            </p>
          </div>
        </div>
        <div
          class="css-7lfzx5"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-11439nx"
    >
      <div
        class="css-j5eaa4"
      >
        <div
          class="css-0"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Frame"
            >
              <path
                d="M15.75 10.5V6C15.75 5.00544 15.3549 4.05161 14.6516 3.34835C13.9484 2.64509 12.9945 2.25 12 2.25C11.0054 2.25 10.0516 2.64509 9.34833 3.34835C8.64507 4.05161 8.24998 5.00544 8.24998 6V10.5M19.606 8.507L20.869 20.507C20.939 21.172 20.419 21.75 19.75 21.75H4.24998C4.09219 21.7502 3.93613 21.7171 3.79193 21.6531C3.64774 21.589 3.51863 21.4953 3.41301 21.3781C3.30738 21.2608 3.2276 21.1227 3.17884 20.9726C3.13008 20.8226 3.11343 20.6639 3.12999 20.507L4.39399 8.507C4.42314 8.23056 4.55361 7.9747 4.76024 7.78876C4.96686 7.60281 5.23501 7.49995 5.51298 7.5H18.487C19.063 7.5 19.546 7.935 19.606 8.507ZM8.62498 10.5C8.62498 10.5995 8.58548 10.6948 8.51515 10.7652C8.44482 10.8355 8.34944 10.875 8.24998 10.875C8.15053 10.875 8.05515 10.8355 7.98482 10.7652C7.91449 10.6948 7.87498 10.5995 7.87498 10.5C7.87498 10.4005 7.91449 10.3052 7.98482 10.2348C8.05515 10.1645 8.15053 10.125 8.24998 10.125C8.34944 10.125 8.44482 10.1645 8.51515 10.2348C8.58548 10.3052 8.62498 10.4005 8.62498 10.5ZM16.125 10.5C16.125 10.5995 16.0855 10.6948 16.0152 10.7652C15.9448 10.8355 15.8494 10.875 15.75 10.875C15.6505 10.875 15.5551 10.8355 15.4848 10.7652C15.4145 10.6948 15.375 10.5995 15.375 10.5C15.375 10.4005 15.4145 10.3052 15.4848 10.2348C15.5551 10.1645 15.6505 10.125 15.75 10.125C15.8494 10.125 15.9448 10.1645 16.0152 10.2348C16.0855 10.3052 16.125 10.4005 16.125 10.5Z"
                id="Vector"
                stroke="#d75951"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </g>
          </svg>
        </div>
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-omcqsq"
          >
            Free Delivery
          </p>
          <p
            class="chakra-text css-0"
          >
            Enter your Postal code for Delivery Availability
          </p>
        </div>
      </div>
      <div
        class="css-7lfzx5"
      />
      <div
        class="css-j5eaa4"
      >
        <div
          class="css-0"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Frame"
            >
              <path
                d="M8.25002 18.75C8.25002 19.1478 8.09198 19.5293 7.81068 19.8107C7.52937 20.092 7.14784 20.25 6.75002 20.25C6.35219 20.25 5.97066 20.092 5.68936 19.8107C5.40805 19.5293 5.25002 19.1478 5.25002 18.75M8.25002 18.75C8.25002 18.3522 8.09198 17.9706 7.81068 17.6893C7.52937 17.408 7.14784 17.25 6.75002 17.25C6.35219 17.25 5.97066 17.408 5.68936 17.6893C5.40805 17.9706 5.25002 18.3522 5.25002 18.75M8.25002 18.75H14.25M5.25002 18.75H3.37502C3.07665 18.75 2.7905 18.6315 2.57952 18.4205C2.36855 18.2095 2.25002 17.9234 2.25002 17.625V14.25M14.25 18.75H16.5M14.25 18.75V14.25M2.25002 14.25V6.61499C2.24843 6.34116 2.34848 6.07648 2.53081 5.87217C2.71313 5.66786 2.96477 5.53845 3.23702 5.50899C6.57006 5.16302 9.92997 5.16302 13.263 5.50899C13.828 5.56699 14.25 6.04699 14.25 6.61499V7.57299M2.25002 14.25H14.25M19.5 18.75C19.5 19.1478 19.342 19.5293 19.0607 19.8107C18.7794 20.092 18.3978 20.25 18 20.25C17.6022 20.25 17.2207 20.092 16.9394 19.8107C16.6581 19.5293 16.5 19.1478 16.5 18.75M19.5 18.75C19.5 18.3522 19.342 17.9706 19.0607 17.6893C18.7794 17.408 18.3978 17.25 18 17.25C17.6022 17.25 17.2207 17.408 16.9394 17.6893C16.6581 17.9706 16.5 18.3522 16.5 18.75M19.5 18.75H20.625C21.246 18.75 21.754 18.246 21.715 17.626C21.5141 14.3236 20.4019 11.1416 18.502 8.43299C18.3211 8.17927 18.0848 7.97001 17.8111 7.82102C17.5373 7.67203 17.2333 7.58722 16.922 7.57299H14.25M14.25 7.57299V14.25"
                id="Vector"
                stroke="#d75951"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </g>
          </svg>
        </div>
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-omcqsq"
          >
            Return Delivery
          </p>
          <p
            class="chakra-text css-0"
          >
            Free 30 days Delivery Return. Details
          </p>
        </div>
      </div>
      <div
        class="css-7lfzx5"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
